apis/apiextensions/v1/composition_transforms.go

```diff
@@ -220,6 +225,11 @@ type StringTransform struct {
 	// Trim the prefix or suffix from the input
 	// +optional
 	Trim *string `json:"trim,omitempty"`
+
+	// Extract first match from the input using a Go regex string. See
+	// https://pkg.go.dev/regexp/ for details.
+	// +optional
+	Regexp *string `json:"regexp,omitempty"`
```

Comment:
You'll need to mirror these changes (just the types, no logic) in `apis/apiextensions/v1alpha1/revision_types.go`. I suspect some changes will also be needed to replicate the values in `internal/controller/apiextensions/composition/revision.go`.

Reply:
Honestly I think we forgot to update it in revision.

Reply:
haha, I'll dig a bit more then

Reply:
That appears to be only copying the `Format` field of the `StringTransform` struct - it's missing the existing `Type` and `Trim` fields, and the new `Regexp` field.

Reply:
Given you're now cleaning up stuff we missed earlier I'd be happy to merge this PR as is and follow up to add this all myself. You will need to figure out the `check-diff` check failing though - typically that means that some change created by running `make reviewable` should be part of this PR but isn't.

Reply:
Sounds like a good option for me, if not too much to ask, can you please somehow show me that follow-up after it's done? would be nice to understand it after all.

I did `make reviewable` magic again, which is indeed has changed some files. So I think it should be good now?